waltz-ng/client/data-types/routes.js

```diff
@@ -0,0 +1,42 @@
+const baseState = {
+    resolve: {
+        dataTypes: [
+            'DataTypesService',
+            (dataTypesService) => dataTypesService.loadDataTypes()
+        ]
+    }
+};
+
+
+const listState = {
+    url: 'data-types',
+    views: {'content@': require('./data-type-list') }
+};
+
+
+const viewByCodeState = {
+    url: 'data-types/code/{code}',
+    views: {'content@': require('./data-type-view') },
+    redirectTo: 'main.data-type.view',
```

Comment:
could we not use a custom resolver for byCodeState and byIdState (viewState) ?

e.g.

````

function byCodeResolver($stateParams, dataTypeService) {
   return {
      dataType:   dataTypeService.findByCode($stateParams.code) 
   };
}

byCodeResovler.$inject = [...]

byCodeState = {
   ....
   resolver: byCodeResolver 
};
````


Then in the controllers you can receive the resolved dataTypes in the injection list.....